### PR TITLE
fix: getContainer works after prev container destory

### PR DESCRIPTION
**Pull request by zuolung**
want to solve the issue from antd[#33781](https://github.com/ant-design/ant-design/issues/33781); 
demo: https://codesandbox.io/s/antd-reproduction-template-forked-2q52f?file=/index.js:242-809

### CODE REVIEW

File: src/Notification.tsx [hunk 1]
```diff
@@ -239,20 +239,35 @@ class Notification extends Component<NotificationProps, NotificationState> {
...
   function ref(notification: Notification) {
     if (called) {
       return;
     }
     called = true;
     callback({
       notice(noticeProps) {
+        if (getContainer && (!mountTarget || !isChildOfDocument(mountTarget))) {
+          mountTarget = getContainer();
+          mountTarget.appendChild(div);
```
Comment (zombieJ) [inline src/Notification.tsx:269]: Seems this code is same as above, could we move into a function instead?

**Comment by zuolung** [comment]
👌，please review

File: src/Notification.tsx (hunk 1 above)
Comment (zuolung) [reply src/Notification.tsx]: 👌，please review

File: src/Notification.tsx [hunk 2]
```diff
@@ -239,20 +239,34 @@ class Notification extends Component<NotificationProps, NotificationState> {
...
-  } else {
-    document.body.appendChild(div);
-  }
+  let mountTarget = document.body;
   let called = false;
+  function isChildOfDocument(child): boolean {
+    if (!child) return false;
+    let parentNode = child.parentNode;
+    while (parentNode) {
+      if (document === parentNode) {
```
Comment (zombieJ) [inline src/Notification.tsx:248]: `document === parentNode || parentNode instanceof Document`

Comment (zuolung) [reply src/Notification.tsx:248]: instanceof just can confirm __proto__

Comment (zuolung) [reply src/Notification.tsx:248]: any other solution？

**Pull request closed without merging**